dask/dataframe/tests/test_dataframe.py

```diff
@@ -2126,6 +2126,42 @@ def test_combine_first():
     assert_eq(ddf1.B.combine_first(df2.B), df1.B.combine_first(df2.B))
 
 
+@pytest.mark.xfail(PANDAS_GE_210, reason="https://github.com/dask/dask/issues/10931")
+@pytest.mark.parametrize(
+    "dtype_lhs,dtype_rhs",
+    [("f8", "i8"), ("f8", "f4"), ("datetime64[s]", "datetime64[ns]")],
+)
+def test_combine_first_all_nans(dtype_lhs, dtype_rhs):
+    """If you call s1.combine_first(s2), where s1 is pandas.Series of all NaNs and s2 is
+    a pandas.Series of non-floats, the dtype becomes that of s2. Starting with pandas
+    2.1, this comes with a deprecation warning.
+    Test behaviour when either a whole dask series or just a chunk is full of NaNs.
```

Comment:
> The new test will break when the FutureWarning changes to implementation changes (pandas 3.0?)

Do you think that either in the xfail we do something like `PANDAS_GE_210 and PANDAS_LT_300` or at least adding something in this test docs about it probably breaking when Pandas 3.0 is out?

Reply:
I thought about `not PANDAS_GE_300`, but I'm not actually sure when the deprecation will be enacted. It's also finicky when pre-releases come into play. I'll add a note.